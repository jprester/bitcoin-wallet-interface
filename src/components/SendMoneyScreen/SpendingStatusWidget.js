import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import { displayFiatValue } from '../../utils/utils';
import styles from './_styles_SendMoneyScreen';

export const SpendingStatusWidget = (props) => {
  const { classes, btcAvailable, btcPrice } = props;

  return (
    <div className={classes.statusWidgetContainer}>
      <h2> My Current Balance </h2>

      <div className={classes.statusPanel}>
        <p className={classes.textLabel}>Bitcoin</p>
        <p className={classes.numberLabel}>{btcAvailable} BTC</p>
      </div>

      <div className={classes.statusPanel}>
        <p className={classes.textLabel}>Balance in usd</p>
        <p className={classes.numberLabel}>{displayFiatValue(btcPrice, btcAvailable)}</p>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  btcAvailable: state.transactionsData.btcAvailable,
  btcSent: state.transactionsData.btcSent,
  btcPrice: state.transactionsData.btcPrice,
});

const styledComponent = withStyles(styles)(SpendingStatusWidget);

export default connect(mapStateToProps)(styledComponent);

SpendingStatusWidget.propTypes = {
  btcAvailable: PropTypes.number.isRequired,
  classes: PropTypes.shape({}).isRequired,
};
