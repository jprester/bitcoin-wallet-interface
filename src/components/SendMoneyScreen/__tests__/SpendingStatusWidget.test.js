import React from 'react';
import { shallow } from 'enzyme';
import { SpendingStatusWidget } from '../SpendingStatusWidget';

const wrapper =
  shallow((<SpendingStatusWidget classes={{}} btcSent={1.2} btcAvailable={10} />));

describe('SpendingStatusWidget', () => {
  it('should render SpendingStatusWidget correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
