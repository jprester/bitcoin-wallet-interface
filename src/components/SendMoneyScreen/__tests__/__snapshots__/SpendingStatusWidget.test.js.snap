// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpendingStatusWidget should render SpendingStatusWidget correctly 1`] = `
<div>
  <h2>
     My Current Balance 
  </h2>
  <div>
    <p>
      Bitcoin
    </p>
    <p>
      10
       BTC
    </p>
  </div>
  <div>
    <p>
      Balance in usd
    </p>
    <p>
      NaN
       $
    </p>
  </div>
</div>
`;
